tests/test_modeling_tf_common.py

```diff
@@ -450,7 +452,16 @@ def test_compile_tf_model(self):
         metric = tf.keras.metrics.SparseCategoricalAccuracy("accuracy")
 
         for model_class in self.all_model_classes:
-            if self.is_encoder_decoder:
+            if model_class.__name__ in ["TFSpeech2TextModel", "TFSpeech2TextForConditionalGeneration"]:
+                inputs = {
+                    "decoder_input_ids": tf.keras.Input(
+                        batch_shape=(2, max_input),
+                        name="decoder_input_ids",
+                        dtype="int32",
+                    ),
+                    "input_ids": tf.keras.Input(batch_shape=(2, 32, max_input), name="input_ids", dtype="float32"),
+                }
```

Comment:
Is there another (more general) check we could use here?

Reply:
By more general do you mean instead of hardcoding the possible names in `model_class.__name__`? We have `TF_MODEL_FOR_SPEECH_SEQ_2_SEQ_MAPPING_NAMES`, but it doesn't cover `TFSpeech2TextModel` :(